### PR TITLE
Sort area types by hierarchy with fallback to total count

### PR DESCRIPTION
**Pull request by thomasridd**
### What
Superficially no differences. If present issues would appear on the Change Area Type screen.

Beneath the hood
* AreaTypes now sorted by the new Hierarchy_order field on AreaType with a fallback to total count
* Shifted sorting code into its own file
* Removed hardcoded sort functions
* Removed ENABLE_CUSTOM_SORT 

### How to review

Sense check
Tests pass

### Who can review

!me

### CODE REVIEW

File: mapper/mapper.go [hunk 1]
```diff
@@ -186,23 +186,8 @@ func CreateAreaTypeSelector(req *http.Request, enableCustomSort bool, basePage c
...
-			Value:       area.ID,
-			Label:       area.Label,
-			Description: area.Description,
-			TotalCount:  area.TotalCount,
-		})
-	}
+	selections := mapAreaTypesToSelection(sortAreaTypes(areaType))
 
-	sort.Slice(selections, func(i, j int) bool {
-		if enableCustomSort {
```
Comment (mr-nick17) [inline mapper/mapper.go:200]: You also need to remove `enableCustomSort` from the `CreateAreaTypeSelector` signature

File: mapper/sorting.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package mapper
+
+import (
+	"sort"
+	"strconv"
+
+	"github.com/ONSdigital/dp-api-clients-go/v2/population"
+)
+
+// sorts population categories by ID - numerically if possible, with negatives listed last
```
Comment (mr-nick17) [inline mapper/sorting.go:10]: comment/go-doc should start with the func name 
```suggestion
// sortCategoriesByID sorts population categories by ID - numerically if possible, with negatives listed last
```

File: mapper/sorting.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		})
+	} else {
+		sort.Slice(sorted, func(i, j int) bool {
+			return sorted[i].ID < sorted[j].ID
+		})
+	}
+	return sorted
+}
+
+// sorts area types by pre-defined heirarchies
```
Comment (mr-nick17) [inline mapper/sorting.go:43]: As above go-doc comment

File: mapper/sorting_test.go [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+import (
+	"testing"
+
+	"github.com/ONSdigital/dp-api-clients-go/v2/population"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestSortCategoriesByID(t *testing.T) {
+
```
Comment (mr-nick17) [inline mapper/sorting_test.go:11]: Unnecessary whitespace

File: mapper/sorting_test.go [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+					expected := []string{"1", "10", "2nd Option"}
+					So(actual, ShouldResemble, expected)
+				})
+			})
+		})
+	})
+}
+
+func TestSortAreaTypes(t *testing.T) {
+
```
Comment (mr-nick17) [inline mapper/sorting_test.go:126]: Unnecessary whitespace

**Review by mr-nick17 — COMMENTED** [review]
The feature flag `enableCustomSort` can be removed from the signature in func `CreateAreaTypeSelector` too. You may want to keep it in `config.go` until its removed from `dp-configs` though

**Comment by thomasridd** [comment]
I've returned the ENABLE_CUSTOM_SORT variable to the config

**Review by mr-nick17 — APPROVED** [review]
